docs/design/Chem Integration.md

```diff
@@ -0,0 +1,66 @@
+
+- Identify what package the dependency came from
+- Update a project file to now reference this from
+
+The expected UX for this will be to implement it as an extension that a user can opt into. As a user, in order to run the Chem integration, they will obtain the extension (ie `UpgradeAssitantChem.zip`) and then run as they normally would:
+
+```
+upgrade-assistant some-project.vbproj --extension .\path\to\UpgradeAssistantChem.zip
+```
+
+### Decoupling the data
+Initially, any data needed will be provided with the extension. This data would be what is used to map the fingerprints to actual packages. They often range around the 100mb-200mb range for useful data, but could be potentially larger if desired.
+
+However, it may be nice to decouple the data from the actual extension itself so they can be updated independently. In order to support his, Upgrade Assistant should provide a way for steps/extensions to provide step-dependent data. This could be something that a step could register it as having, and then Upgrade Assistant could provide commands such as the following:
+
+- `upgrade-assistnat manage-data list`: List registered data sources
+- `upgrade-assistant manage-data update --name [name]`: Update a registered data source (managed by the step itself)
+
+There are potentially many questions here as to handle the data. This is provided solely as a way to demonstrate a potential way to decouple the data required. Initially, the data will be included with the extension, but potentially have optional indexes (or indexes that carry more data and are much larger) will require some sort of way to manage the data.
+
+## Challenges
+
+There are a number of challenges with this, as it is completely based on heuristics in order to match package identities. Potential limitations:
+
+- If a package was never released to NuGet, it will never show up in the available indexes
+- Private builds of projects may not match external builds
+- Currently only planning on supporting packages from NuGet.org so loose assemblies from a company's internal teams will not be found.
+
+## Planned Work
+
+Below are the main work items that will be explored broken into two sprints. There is currently a pre-analysis migration effort being worked on, and this work may be updated to ensure that the information flows as expected into the reports generated there as those features become available.
+
+### Sprint 1
+- As a user, I can consume an extension that integrates chem (#482)
+- As a user, the chem integration can identify what loose assemblies I have (#483)
+- As a user, the chem integration will replace my loose assemblies with ones it identifies (#484)
+- As a user, I know which loose assemblies do not have any available matches (#485)
+
+### Sprint 2
+- As a user, I can update my chem data index independently of the extension (#486)
+- As a user, I can have access to the same data from multiple projects (#487)
```

Comment:
What does this mean?

Reply:
I've reworded it, but it's about being able to manage data at a global level in upgrade assistant (ie simplify storage of indexes so they can easily be consumed/updated/etc)